dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowFactoryImpl.java

```diff
@@ -924,16 +929,17 @@ public void saveAction(final WorkflowAction action) throws DotDataException,Alre
 			action.setId(UUIDGenerator.generateUuid());
 		}
 
-		final DotConnect db = new DotConnect();
+		final String     nextStep = this.getNextStep(action);
+		final DotConnect db       = new DotConnect();
 		if (isNew) {
 			db.setSQL(sql.INSERT_ACTION);
 			db.addParam(action.getId());
 			db.addParam(action.getSchemeId());
 			// we are not longer using the stepId, the relationship now is with schemeId, however it needs a step id to work
-			db.addParam(UtilMethods.isSet(action.getStepId())?action.getStepId():action.getNextStep());
+			//db.addParam(UtilMethods.isSet(action.getStepId())?action.getStepId():nextStep);

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

Reply:
Done